drivers/sensor/tdk/icm456xx/icm456xx.c

```diff
@@ -0,0 +1,1061 @@
+			val->val1 = cfg->apex;
+		}
+		break;
+
+	default:
+		LOG_ERR("Unsupported channel");
+		res = -EINVAL;
+		break;
+	}
+
+	icm456xx_unlock(dev);
+
+	return res;
+}
+
+static inline int icm456xx_bus_check(const struct device *dev)
+{
+	const struct icm456xx_config *cfg = dev->config;
+
+	return cfg->bus_io->check(&cfg->bus);
+}
+
+static int icm456xx_init(const struct device *dev)
+{
+	struct icm456xx_data *data = dev->data;
+	int res = 0;
+
+	if (icm456xx_bus_check(dev) < 0) {
+		LOG_ERR("bus check failed");
+		return -ENODEV;
+	}
+
+	data->accel_x = 0;
+	data->accel_y = 0;
+	data->accel_z = 0;
+	data->gyro_x = 0;
+	data->gyro_y = 0;
+	data->gyro_z = 0;
+	data->temp = 0;
+
+	if (icm456xx_sensor_init(dev)) {
+		LOG_ERR("could not initialize sensor");
+		return -EIO;
+	}
+
+#ifdef CONFIG_ICM456XX_TRIGGER
+	res |= icm456xx_trigger_enable_interrupt(dev);
+	res |= icm456xx_trigger_init(dev);
+	if (res < 0) {
+		LOG_ERR("Failed to initialize interrupt.");
+		return res;
+	}
+#endif
+
+	res |= icm456xx_turn_on_sensor(dev);
+
+	return res;
+}
+
+#ifndef CONFIG_ICM456XX_TRIGGER
+
+void icm456xx_lock(const struct device *dev)
+{
+	ARG_UNUSED(dev);
+}
+
+void icm456xx_unlock(const struct device *dev)
+{
+	ARG_UNUSED(dev);
+}
+
+#endif
+
+static DEVICE_API(sensor, icm456xx_driver_api) = {
+#ifdef CONFIG_ICM456XX_TRIGGER
+	.trigger_set = icm456xx_trigger_set,
+#endif
+	.sample_fetch = icm456xx_sample_fetch,
+	.channel_get = icm456xx_channel_get,
+	.attr_set = icm456xx_attr_set,
+	.attr_get = icm456xx_attr_get,
+};
+
+/* device defaults to spi mode 0/3 support */
+#define ICM456XX_SPI_CFG (SPI_OP_MODE_MASTER | SPI_WORD_SET(8) | SPI_TRANSFER_MSB | SPI_MODE_CPOL | SPI_MODE_CPHA)
+
+/* Initializes a common struct icm456xx_config */
+#define ICM456XX_CONFIG_COMMON(inst)                                                               \
+	IF_ENABLED(CONFIG_ICM456XX_TRIGGER,	\
+			(.gpio_int = GPIO_DT_SPEC_INST_GET_OR(inst, int_gpios, {0}),))     \
+			.accel_fs = DT_INST_ENUM_IDX(inst, accel_fs),	\
+			.accel_hz = DT_INST_ENUM_IDX(inst, accel_hz),	\
+			.accel_pwr_mode = DT_INST_ENUM_IDX(inst, power_mode),	\
+			.accel_avg = DT_INST_ENUM_IDX(inst, accel_avg),	\
+			.accel_filt_bw = DT_INST_ENUM_IDX(inst, accel_filt_bw_hz),	\
+			.apex = DT_INST_ENUM_IDX(inst, apex),                          \
+			.gyro_fs = DT_INST_ENUM_IDX(inst, gyro_fs),                     \
+			.gyro_hz = DT_INST_ENUM_IDX(inst, gyro_hz),                     \
+			.gyro_filt_bw = DT_INST_ENUM_IDX(inst, gyro_filt_bw_hz)
+
+/* Initializes the bus members for an instance on a SPI bus. */
+#define ICM456XX_CONFIG_SPI(inst)                                                                  \
+	{.bus.spi = SPI_DT_SPEC_INST_GET(inst, ICM456XX_SPI_CFG, 0),                               \
+	 .bus_io = &icm456xx_bus_io_spi,                                                           \
+	 .serif_type = UI_SPI3,                                                                    \
```

Comment:
I would use SPI4 as more standard communication. Does your platform only support SPI3? ANy I think this is not used as you don't call the init() API.

Reply:
Modified to SPI4 and tested